project-base/storefront/components/Layout/Header/Cart/CartInHeader.tsx

```diff
@@ -38,97 +45,145 @@ export const CartInHeader: FC = ({ className }) => {
+                        !cart?.items.length
+                            ? 'lg:flex lg:w-96 lg:flex-nowrap lg:items-center lg:justify-between'
+                            : 'lg:w-[548px]',
+                    )}
+                >
+                    {(isHovered || isClicked) && (
+                        <>
+                            {cart?.items.length ? (
+                                <>
+                                    <div className="flex flex-row justify-between mb-10 lg:hidden pr-1">
+                                        <ExtendedNextLink href={cartUrl} onClick={() => setIsClicked(false)}>
+                                            <ArrowRightIcon className="rotate-180 text-borderAccent w-4" />
+                                        </ExtendedNextLink>
+                                        <span className="text-base">{t('Cart')}</span>
+                                        <RemoveIcon
+                                            className="w-4 text-borderAccent cursor-pointer"
+                                            onClick={() => setIsClicked(false)}
+                                        />
+                                    </div>
+                                    <ul className="relative w-[315px] max-h-[78dvh] m-0 flex list-none flex-col overflow-y-auto p-0 lg:w-[510px] lg:max-h-[50dvh] overflow-auto">
+                                        {isRemovingFromCart && <LoaderWithOverlay className="w-16" />}
+                                        {cart.items.map((cartItem, listIndex) => (
+                                            <CartInHeaderListItem
+                                                key={cartItem.uuid}
+                                                cartItem={cartItem}
+                                                onRemoveFromCart={() => removeFromCart(cartItem, listIndex)}
+                                            />
+                                        ))}
+                                    </ul>
+                                    <div
+                                        className={twJoin(
+                                            'flex pt-5 gap-4',
+                                            shouldDisplayTransportBar ? 'justify-between' : 'justify-end',
+                                        )}
+                                    >
+                                        <FreeTransportRange />
+                                        <Button
+                                            className="rounded-lg"
+                                            size="small"
+                                            onClick={() => router.push(cartUrl)}
+                                        >
+                                            {t('Go to cart')}
+                                        </Button>
+                                    </div>
+                                </>
+                            ) : (
+                                <>
```

Comment:
Do we need an empty popup? The button already indicates that the shopping cart is empty.

Reply:
In some project we have made it is present. I think it is project by project choice. So I think it will be easier to remove on project.